clients/python/tests/unit/serde/json_serde_test.py

```diff
@@ -0,0 +1,55 @@
+import unittest
+
+from raccoon_client.protos.raystack.raccoon.v1beta1.raccoon_pb2 import SendEventRequest, Event, SendEventResponse, Status, Code
+from raccoon_client.serde.json_serde import JsonSerde
+from tests.unit.rest.client_test import get_static_uuid, get_static_time_ns, get_stub_response_json
+
+
+def get_event_request():
+    request = SendEventRequest()
+    request.req_guid = get_static_uuid()
+    request.sent_time.FromNanoseconds(get_static_time_ns())
+    e = Event()
+    e.type = "topic 1"
+    e.event_bytes = b'{"random1": "abc", "xyz": 1}'
+    request.events.append(e)
+    return request
+
+
+class JsonSerdeTest(unittest.TestCase):
+    serde = JsonSerde()
+    marshalled_event_request = """{
+  "reqGuid": "17e2ac19-df8b-4a30-b111-fd7f5073d2f5",
+  "sentTime": "2023-08-17T05:38:49.234986Z",
+  "events": [
+    {
+      "eventBytes": "eyJyYW5kb20xIjogImFiYyIsICJ4eXoiOiAxfQ==",
+      "type": "topic 1"
+    }
+  ]
+}"""
+    serialised_event_request = b'{\n  "reqGuid": "17e2ac19-df8b-4a30-b111-fd7f5073d2f5",\n  "sentTime": "2023-08-17T05:38:49.234986Z",\n  "events": [\n    {\n      "eventBytes": "eyJyYW5kb20xIjogImFiYyIsICJ4eXoiOiAxfQ==",\n      "type": "topic 1"\n    }\n  ]\n}'
+
+    def test_serialise_of_input(self):
+        event = {"random1": "abc", "xyz": 1}
+        self.assertEqual(self.serde.serialise(event), b'{"random1": "abc", "xyz": 1}')
+
+    def test_serialise_of_proto_object(self):
+        event = get_event_request()
+        serialised_proto = self.serde.serialise(event)
+        self.assertEqual(serialised_proto, self.serialised_event_request)
+
+    def test_marshaling_of_proto_message(self):
+        request = get_event_request()
+        self.assertEqual(self.serde.marshal(request), self.marshalled_event_request)
+
+    def test_unmarshalling_of_proto_message(self):
```

Comment:
this test if for unmarshalling of json message, right?

Reply:
can we add tests for marshal of json?

Reply:
I didnt include marshal and unmarshal of json/dict because marshal is always going to act on proto object SendEventRequest and SendEventRespone.

Reply:
okay, but the test name can be improved I guess, it is actually taking a json string instead of a proto message right?

Reply:
Do you mean it's application/json content type and not application/proto ?
I think I meant it to be test_unmarshalling_into_proto_message, will updated the name.

Reply:
Done